exec/internal/compile/compile.go

```diff
@@ -131,16 +137,38 @@ type block struct {
 	branchTables []*BranchTable   // All branch tables that were defined in this block.
 }
 
+// BytecodeMetadata encapsulates metadata about a bytecode stream.
+type BytecodeMetadata struct {
+	BranchTables []*BranchTable
+	Instructions []InstructionMetadata
+
+	// Inbound jumps - used by the AOT/JIT scanner to
+	// avoid generating native code which has an inbound
+	// jump target somewhere deep inside.
+	InboundTargets map[int64]bool
```

Comment:
depending on the number of elements this map may hold, we might consider using `map[int64]struct{}` instead.